evaluations/rollups.md

```diff
@@ -0,0 +1,219 @@
+| --------------------- | ------------- | --------------- | ------------ |
+| Fuel L2 V1            | Yes           | Yes             | Yes          |
+| Metis                 | Yes           | No              | No           |
+| Taiko                 | Yes           | No              | No           |
+| Arbitrum              | No            | N/A             | N/A          |
+| Base                  | No            | N/A             | N/A          |
+| Blast                 | No            | N/A             | N/A          |
+| Boba network          | No            | N/A             | N/A          |
+| Fuel Rollup OS        | No            | N/A             | N/A          |
+| Immutable X           | No            | N/A             | N/A          |
+| Immutable zkEVM       | No            | N/A             | N/A          |
+| Linea                 | No            | N/A             | N/A          |
+| Mantle                | No            | N/A             | N/A          |
+| Optimism              | No            | N/A             | N/A          |
+| Polygon zkEVM         | No            | N/A             | N/A          |
+| Polygon Miden         | No            | N/A             | N/A          |
+| Scroll                | No            | N/A             | N/A          |
+| Starknet              | No            | N/A             | N/A          |
+| zkSync lite           | No            | N/A             | N/A          |
+| zkSync Era            | No            | N/A             | N/A          |
+
+Conclusion
+----------
+
+There seems to be no rollup that matches all the requirements that we listed in
+the beginning of the document. The most pressing problem is that only Mantle
+seems to be cheap enough to allow storage providers to turn a profit, given the
+assumptions of an average 10 GB slot size and 1 proof per 24 hours. It is
+unclear whether these low prices are sustainable in the long run. If we want to
+have more choice on where to deploy, then we either need to reduce the number of
+on-chain proofs in Codex drastically, or we need to find a way to reduce rollup
+transaction costs.
+
+Luckily we're already working on reducing the number of proofs by introducing
+proof aggregation, but the analysis in this document shows that we might not be
+able to launch  a storage network without it. Reducing the number of proofs also
+ensures that the network can grow to a larger total size.
+
+When we look at reducing the transaction costs, then the best thing to focus on
+is getting rid of the need to post transactions on L1 in blobs. This is by far
+the most expensive part of running a rollup, and this is most likely also why
+Mantle is the only rollup in this overview that is cheap enough; it uses EigenDA
+instead of posting to L1. In this respect, it might also be interesting to look
+at Arbitrum AnyTrust, which has a similar design. We could also consider
+creating a fork of an existing rollup and use Codex as the DA layer. Some of the
```

Comment:
Wouldn't this be a bit of a circular issue? Codex relies on the rollup to work, and the rollup relies on Codex to work.

Reply:
I think that you can do it like this (rough idea):
1. rollup creates block number x
2. rollup creates block number x+1, and includes a storage request for block x
3. rollup creates more blocks....
4. the storage request for block x is started, all it slots are filled
5. validators can now check block x...
6. if after 7 days no fraud proofs were detected, then block x is considered final

If you use want to use zk proofs of the block instead of fraud proofs, then you can consider block x final at step 4.